File: 백준/Bronze/2745. 진법 변환/진법 변환.cc
#include <iostream>
#include<cmath>

using namespace std;

int main() {
    string s;
    int N;
    int ans = 0;
    cin >> s >> N;
        
    for (int i = 0; i < s.length(); i++) {
        if (s[i] >= '0' && s[i] <= '9') {
            ans += (s[i] - '0') * pow(N, s.length() - 1 - i);
        } else if (s[i] >= 'A' && s[i] <= 'Z')
            ans += (s[i] - 55) * pow(N, s.length() - 1 - i);
        }
    
    cout << ans;
    return 0;
}

File: 백준/Bronze/2745. 진법 변환/진법 변환.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        
        String s = st.nextToken();
        int N = Integer.parseInt(st.nextToken());
        int ans = 0;
        
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) >= '0' && s.charAt(i) <= '9') {
                ans += (s.charAt(i) - '0') * Math.pow(N, s.length() - 1 - i);
            } else if (s.charAt(i) >= 'A' && s.charAt(i) <= 'Z')
                ans += (s.charAt(i) - 55) * Math.pow(N, s.length() - 1 - i);
        }
        
        System.out.print(ans);
    }
}